File: backend/app/schemas/checkout.py
from pydantic import BaseModel
from uuid import UUID


class CreateOrderRequest(BaseModel):

    merchant_id: UUID

    amount: float


class CreateOrderResponse(BaseModel):

    order_id: str

    amount: int

    currency: str

    key: str


class VerifyPaymentRequest(BaseModel):

    merchant_id: UUID

    recommendation_id: UUID | None = None

    razorpay_order_id: str

    razorpay_payment_id: str

    razorpay_signature: str

File: frontend/services/checkout.ts
import api from "./api";

export async function createOrder(
  merchantId: string,
  amount: number,
) {
  const { data } = await api.post(
    "/checkout/create-order",
    {
      merchant_id: merchantId,
      amount,
    },
  );

  return data;
}

export async function verifyPayment(
  payload: {
    merchant_id: string;
    recommendation_id?: string;
    razorpay_order_id: string;
    razorpay_payment_id: string;
    razorpay_signature: string;
  },
) {
  await api.post("/checkout/verify", payload);
}